docs/specs/code/LOCK_WALLET_WITH_PIN.md

```diff
@@ -0,0 +1,41 @@
+# Abstract
+
+Allows user to lock their wallet with a pin like someone can do in iOS / Android.
+
+# Motivation
+
+Yoroi is completely secure (impossible to do tx without the password), but we want to add a lock screen similar to what phones have so we can protect the user privacy (balance and transactions) in case that someone else has access to the computer. This has a been a constant request from multiple people.

```

Comment:
AFAIK 2FA is token based and it’s used when there is a server that can verified the token. We don’t have a server, so I feel if we make it work it would be pretty hacky.

Reply:
The stuff like Google 2FA works with a shared secret, so we potentially could do this, but the damn thing uses HMAC which is symmetrical, so we would have some kind of an encryption to store the shared secret itself [facepalm.jpg] 

If we can find a 2FA service that uses asymmetric key-pairs  - then we could easily keep our public key in open, if app guarantees that it keeps its key in secret.

Reply:
https://blog.trezor.io/why-you-should-never-use-google-authenticator-again-e166d09d4324

Reply:
> https://blog.trezor.io/why-you-should-never-use-google-authenticator-again-e166d09d4324
> 
> "FIDO/U2F" looks exactly right, cuz user (the 2FA app) owns the secret key, and Yoroi storage (server in this case) would only contain the public key, which is non-critical to be leaked.

Reply:
But I reckon this could be a separate feature/project